pkg/console/subresource/manifestwork/oauth-client-work.go

```diff
@@ -0,0 +1,85 @@
+package manifestwork
+
+import (
+	"errors"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/console-operator/pkg/api"
+	oauthclientsub "github.com/openshift/console-operator/pkg/console/subresource/oauthclient"
+	"github.com/openshift/console-operator/pkg/console/subresource/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	workv1 "open-cluster-management.io/api/work/v1"
+)
+
+func DefaultManagedClusterOAuthClientManifestWork(operatorConfig *operatorv1.Console, namespace string, secret string, redirects []string) *workv1.ManifestWork {
+	manifestWork := ManagedClusterOAuthClientManifestWorkStub(namespace)
+	oauthClient := oauthclientsub.DefaultManagedClusterOauthClient(secret, redirects)
+	WithManifest(manifestWork, oauthClient)
+	util.AddOwnerRef(manifestWork, util.OwnerRefFrom(operatorConfig))
+	return manifestWork
+}
+
+func ManagedClusterOAuthClientManifestWorkStub(namespace string) *workv1.ManifestWork {
+	return &workv1.ManifestWork{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      api.ManagedClusterOauthClientManifestWork,
+			Namespace: namespace,
+			Labels:    util.LabelsForManagedClusterResources(""),
+		},
+		Spec: workv1.ManifestWorkSpec{
+			Executor: &workv1.ManifestWorkExecutor{
+				Subject: workv1.ManifestWorkExecutorSubject{
+					Type: workv1.ExecutorSubjectTypeServiceAccount,
+					ServiceAccount: &workv1.ManifestWorkSubjectServiceAccount{
+						Namespace: api.OpenShiftConsoleOperatorNamespace,
+						Name:      api.OpenShiftConsoleOperatorExecutor,
+					},
+				},
+			},
+			Workload: workv1.ManifestsTemplate{
+				Manifests: []workv1.Manifest{},
+			},
+		},
+	}
+}
+
+func WithManifest(manifestWork *workv1.ManifestWork, object runtime.Object) {
+	if &manifestWork.Spec == nil {
```

Comment:
dont think these checks are necessary since we are always using the stub from the `ManagedClusterOAuthClientManifestWorkStub` which should always contain all those fields

Reply:
I'll probably be removing this logic in preference to the workbuilder helper here: https://github.com/open-cluster-management-io/api/tree/main/utils/work/v1